backends/SharedDummy/src/SharedMemoryManager.cc

```diff
@@ -5,6 +5,42 @@
 
 namespace ChimeraTK {
 
+  namespace {
+    // Proxy class to add try_lock_for() to boost::interprocess::named_mutex::named_mutex, which is not present on older
+    // BOOST versions like on Ubuntu 20.04. This can be removed and replaced with boost::interprocess::named_mutex once
+    // we are fully on Ubuntu 24.04 or newer.
+    class IpcNamedMutex {
+     public:
+      explicit IpcNamedMutex(boost::interprocess::named_mutex& mx) : _mx(mx) {}
+
+      template<typename Duration>
+      // NOLINTNEXTLINE(readability-identifier-naming)
+      bool try_lock_for(const Duration& dur) {
+        auto abs_time = boost::posix_time::microsec_clock::local_time();
```

Comment:
this needs to be universal_time()   (I tried it out since docu is not clear)

Reply:
If I am not mistaken, both universal and local time can have leap seconds and run backwards, so you can  have times "in the past" and the same time twice.
We should use a steady clock. However, I don't know if that's consistent across multiple processes.

Reply:
@drothe, I have tested this code to work on both focal and noble. Maybe I did not check everything on focal, but on noble the timeout definitively works as expected, and so does the in-place construction of the mutex. Can you be more specific in which way it does not work for you?